eng/Testing.targets

```diff
@@ -1,3 +1,92 @@
 <Project>
 
+  <!--
+      Environment variables:
+        - IsGitHubActionsRunner:       indicates whether tests are currently run on GitHub Actions; computed, overridable
+        - IsAzdoCIRunner:              indicates whether tests are currently run on Azure DevOps; computed, overridable
+        - IsAzdoHelixRunner:           indicates whether tests are currently run on Helix; computed, overridable
+
+      Project capabilities:
+        - IsTestProject:               indicates whether the project is a test project; default is false; computed, overridable
+        - IncludeTestUtilities:        indicates whether the test project must not include the TestUtilities project reference; default is false; overridable
+
+      Project requirements:
+        - RunOnGithubActions:          indicates whether tests should run on GitHub Actions (either Windows or Linux); computed
+        - RunOnGithubActionsWindows:   indicates whether tests should run on Windows in GitHub Actions; default is true; overridable
+        - RunOnGithubActionsLinux:     indicates whether tests should run on Linux in GitHub Actions; default is true; overridable
+        - RunOnAzdoCI:                 indicates whether tests should run on Azure DevOps; default is true; overridable
+        - RunOnAzdoHelix:              indicates whether tests should run on Helix (either Windows or Linux); computed
+        - RunOnAzdoHelixWindows:       indicates whether tests should run on Windows in Helix; default is true; overridable
+        - RunOnAzdoHelixLinux:         indicates whether tests should run on Linux in Helix; default is true; overridable
+    -->
+
+  <PropertyGroup>
+    <!-- See https://docs.github.com/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables -->
+    <IsGitHubActionsRunner Condition=" '$(IsGitHubActionsRunner)' == '' and '$(GITHUB_ACTIONS)' == 'true' ">true</IsGitHubActionsRunner>
+
+    <!-- See https://learn.microsoft.com/azure/devops/pipelines/build/variables#system-variables -->
+    <IsAzdoCIRunner Condition=" '$(IsAzdoCIRunner)' == '' and '$(SYSTEM_TEAMPROJECT)' != '' ">true</IsAzdoCIRunner>
+
+    <IsAzdoHelixRunner Condition=" '$(IsAzdoHelixRunner)' == '' and '$(PrepareForHelix)' == 'true' ">true</IsAzdoHelixRunner>
```

Comment:
nit: We don't need to check whether the property is empty or not. We are setting the value based on some environment properties. And if for a particular run we want to force which runner it is then that property could be set on the command line thus completely overriding these.

Reply:
We do to be able simulate different runs locally.